### PR TITLE
New constants and small code improvements

### PR DESCRIPTION
**Pull request by susgo**

### CODE REVIEW

**Comment by susgo** [comment]
Feel free to merge this or close it.

File: src/Csv.php [hunk 1]
```diff
@@ -331,13 +332,35 @@ class Csv {
      * Constructor
      * Class constructor
      *
-     * @param  string|null  $input          The CSV string or a direct filepath
-     * @param  integer|null $offset         Number of rows to ignore from the beginning of  the data
-     * @param  integer|null $limit          Limits the number of returned rows to specified amount
-     * @param  string|null  $conditions     Basic SQL-like conditions for row matching
-     * @param  null|true    $keep_file_data Keep raw file data in memory after successful parsing (useful for debugging)
+     * @param  string|null $input The CSV string or a direct filepath
+     * @param  integer|null $offset Number of rows to ignore from the beginning
```
Comment (Fonata) [inline src/Csv.php]: The formatting seems to have changed without added benefit. I suspect you need to enable 'Align tag comments' in PhpStorm:

image

I've zipped a fresh copy of my `.idea` folder into #101. If you think that helps: [IntelliJ.and.PhpStorm.settings.zip](https://github.com/parsecsv/parsecsv-for-php/files/1755896/IntelliJ.and.PhpStorm.settings.zip)

File: src/Csv.php [hunk 2]
```diff
@@ -332,13 +332,35 @@ class Csv {
...
+     *   of  the data
+     * @param  integer|null $limit Limits the number of returned rows to
+     *   specified amount
+     * @param  string|null $conditions Basic SQL-like conditions for row
+     *   matching
+     * @param  null|true $keep_file_data Keep raw file data in memory after
+     *   successful parsing (useful for debugging)
      */
-    public function __construct($input = null, $offset = null, $limit = null, $conditions = null, $keep_file_data = null) {
+    public function __construct($input = NULL, $offset = NULL, $limit = NULL, $conditions = NULL, $keep_file_data = NULL) {
```
Comment (Fonata) [inline src/Csv.php]: Groß- und Kleinschreibung: Von Drupal bin ich `NULL` gewöhnt, von RDO und diesem Projekt `null`. Hast du es mit Absicht geändert? Gibt es einen Standard? 

Mir ist es ziemlich gleich welche Variante, doch ist Konsistenz innerhalb des Projektes wichtig.

Comment (susgo) [reply src/Csv.php]: Sorry, I think that comes from our project. I think standard is lowercase and I prefer this too. Will change it.

File: src/Csv.php (hunk 1 above)
Comment (Fonata) [reply src/Csv.php]: I didn't even know you could import the settings. I just replace the folder after comparing it with e.g. WinMerge. For other projects, we selectively commit handpicked files from the `.idea` folder, including the complete `codeStyles` folder. Everything else goes to `.gitignore` so no-one accidentally commits it.

Comment (susgo) [reply src/Csv.php]: done

File: src/Csv.php (hunk 2 above)
Comment (susgo) [reply src/Csv.php]: done

**Comment by susgo** [comment]
Okay, copied your settings folder and reformated the whole file. Should be ready for merge now? Or have I missed something?

File: src/Csv.php [hunk 3]
```diff
@@ -379,32 +399,29 @@ public function parse($input = null, $offset = null, $limit = null, $conditions
             $input = $this->file;
         }
 
-        if (!empty($input)) {
-            if (!is_null($offset)) {
-                $this->offset = $offset;
-            }
+        if (empty($input)) {
+            // todo: but why true?
+            return true;
```
Comment (Fonata) [inline src/Csv.php]: You're right about the todo. The return value used to be `false` if `$this->data === false` - but that is weird, as a second, unsuccessful call to `parse` would still return `true`. So in short, it should return `false` here. I'll add a test for both cases when I merge it.

**Comment by Fonata** [comment]
Thanks, @susgo!

@jimeh @williamknauss This pull request is ready for merging from my perspective. Feel free to comment.

If I don't here back from you, I'll merge this in 7 days at the earliest, which is 7 March.

There is still the issue of the return value of the `parse` function. It was weird before, and is weird now in a different way. That is not blocking the pull request in my opinion.

**Pull request merged**